### DISCUSSION TITLE
o3d.geometry.get_rotation_matrix_from_axis_angle([0., 0., 0.]) counterintuitive

### DISCUSSION THREAD

**Discussion by JanKovis** [isl-org/Open3D#3655] [General]
This no-rotation returns a matrix full of nan-s.
I think less surprising and correct approach would be to return identity matrix - we do no-rotation on every axis, it would be somewhat more sensible result than to do the zero-rotation check on application side. Or at least the behavior might be documented.

Implementation utilizes Eigen and angle normalization which fails for null-rotation probably

Eigen::Matrix3d Geometry3D::GetRotationMatrixFromAxisAngle(
        const Eigen::Vector3d& rotation) {
    const double phi = rotation.norm();
    return Eigen::AngleAxisd(phi, rotation / phi).toRotationMatrix();
}

**Comment by germanros1987** [comment]
In 0.17 I do get the identity matrix.

**Reply by JanKovis** [reply]
Mine was 0.16, obviously it got fixed in meantime

**Comment by saurabheights** [comment]
Is your system arm64? Thinking it might be related to another GitHub issue. https://github.com/isl-org/Open3D/issues/6301

**Reply by JanKovis** [reply]
No, I used X86_64 Windows, if I recall correctly.
